elements/pl-xss-safe/pl-xss-safe.mustache

```diff
@@ -0,0 +1,14 @@
+<script>
+(function() {
+    let code = {{{quoted_code}}};
+    let language = "{{language}}";
+    if (language == "markdown") {
+        let renderer = new showdown.Converter({
+            'literalMidWordUnderscores':true,
+            'literalMidWordAsterisks':true,
+        });
+        code = renderer.makeHtml(code);
+    }
+    document.write(filterXSS(code));
```

Comment:
Where is the `filterXSS` function coming from?

Reply:
Can you help me understand this `document.write()` here? Does that replace the entire document? Or how does it know where to insert/replace?

Reply:
> Where is the `filterXSS` function coming from?

Same as `showdown` comment above.

> Can you help me understand this `document.write()` here? Does that replace the entire document? Or how does it know where to insert/replace?

`document.write()` inserts the HTML in the current position in the file. Note that the function is called immediately while page loading is still in progress, not after loading is complete.

Reply:
Great! Thanks for the explanation.